README.md

```diff
@@ -7,84 +7,72 @@ Dyanmic Hypercore encryption provider
 ```js
 const HypercoreEncryption = require('hypercore-encryption')
 
-const encryption = new HypercoreEncryption({
-  blindingKey,
-  getBlockKey (id, contexxt) {
-    // get key info corresponding to id and context...
-
-    return {
-      version, // encryption scheme
-      padding, // padding byte length
-      key // block key
-    }
-  }
-})
+const getEncryptionKey = async (id) => {
+  // get key info corresponding to id...
 
-const core = new Hypercore(storage, { encryption })
-await core.ready()
+  return {
+    id, // encryption scheme
+    payload // encryption key payload
+  }
+}
 
-await core.append('encrypt with key 1')
+const encryption = new HypercoreEncryption(getEncryptionKey)
 
-await encryption.load(99)
+const core = new Hypercore(storage, {
+  encryption: encryption.createEncryptionProvider({
+    transform (entropy) {
+      return crypto.hash([NAMESPACE, entropy]) // optionally hash
+    }
+  })
+})
 
-await core.append('encrypt with key 99')
+await core.ready()
+await core.append('encrypt with key')
 ```
 
 ## API
 
-#### `const enc = new HypercoreEncryption({ blindingKey, getBlockKey, getBlindingKey })`
+#### `const enc = new HypercoreEncryption(getEncryptionKey)`
 
-Instantiate a new encryption provider. Optionally pass a `preopen` promise that resolves to a key id to be loaded initially.
+Instantiate a new encryption provider.
 
-Provide a hooks with the signature:
+Takes a hook with the signature:
 ```js
-function getBlockKey (id, context) {
-  // context provides information about the core, eg:
-  //   context.key
-  //   context.manifest
-
-  // id id is passed as -1, the module expects the key to be updated
+function getEncryptionKey (id) {
+  // if id is passed as -1, the module expects the latest key
 
   return {
-    version, // encryption scheme
-    padding, // padding byte length
-    key // block key
+    id, // encryption id
+    entropy // entropy
   }
 }
 
-function getBlockKey (context) {
-  return blindingKey // 32 byte blinding key
-}
-
 ```
 
-#### `const padding = enc.padding(context)`
-
-The number of padding bytes.
-
-#### `enc.seekable`
-
-Boolean on whether the current scheme allows for seeks.
-
-#### `enc.version`
-
-The version of the currently loaded scheme.
+#### `const provider = enc.createEncryptionProvider({ transform, compat })`
 
-#### `await enc.load(id, context)`
+Create an encryption provider.
 
-Load the key under `id` and set to be the current encryption info.
-
-#### `await enc.encrypt(index, block, fork)`
-
-Encrypt a block in place.
+```
+{
+  transform (ctx, entropy) {
+    // implement custom block key derivation
+    // entropy will be passed as null when a compat
+    // is expected
+  },
```

Comment:
Missing flag arg for whether `compat` and should be marked as `async`.

For the body, should it mention that it can generate a `blinding` key and a `hash`? Maybe noting it depends if `compat` or not. Also would be helpful to showcase the shape of the return object:
```js
{
  block,
  hash,
  blinding
}
```